### ISSUE TITLE
[NEED HELP]

### ISSUE THREAD

**Issue by malzuraga** [theCrius/project-zomboid-random-zombies-full-plugin-template#2]
**Is your feature request related to a problem? Please describe.**
compatibility patch

**Describe the idea you have**
if I want to add compatibility for the Bandits mod for my friends and I (Unless I have permission to make a public one, or you want to make one?) Which .lua's do I add local isBandit = zombie:getVariableBoolean("Bandit") to?

**Additional context**
I also posted this in the steam discussion..then realized it said to reach out here.

**Comment by theCrius** [comment]
Can you elaborate a bit further about what this "Bandit" mod does? I assume that it uses zombies as entity to create "bandits" on the map. What do you want this mod to do? Ignore the changes on some specific type of check/change in case there is a "bandit" mod in the mod list? Something else?

**Comment by malzuraga** [comment]
https://steamcommunity.com/sharedfiles/filedetails/?id=3268487204&searchtext=bandits is the link.
will copy and paste some stuff from there.

Compatibility
Bandits mod codebase is quite big and unusual. It introduces many hacks and workarounds to evade limits of the game. Incompatibilities may arise with other mods as side effects. If you run into problems, it's recommended to add this mod first and then gradually add others and test along the way.

Zombie Mods
Generally mods that alter zombies in some way may be incompatible with Bandits mod. This is because Bandits are technically zombies. Some modders have already added compatibility with Bandits by explicitly excluding zombie bandits from alteration they would normally do to zombies.

Mods that were adapted to work with Bandits
[CDDA Zomboids](steam://openurl/https://steamcommunity.com/sharedfiles/filedetails/?id=2914016243)
[The Last of Us Infected](steam://openurl/https://steamcommunity.com/sharedfiles/filedetails/?id=3248766883)

If you are a modder and you wish to recognize bandits from other zombies, here is the proper way to do it:
local isBandit = zombie:getVariableBoolean("Bandit")
where zombie is an instance of IsoZombie class.




Recognized Incompatibilities
Tripping zombies - it works but the bandits trip too
Weapon Condition Indicator - works but will throw an irrelevant error. Use popular alternatives such as [Item Condition](steam://openurl/https://steamcommunity.com/sharedfiles/filedetails/?id=2852309899&searchtext=condition+indicator)
Don't feed the corpses - bandits will eat zombies :)
**Random zombies** - _it will work but bandits will carry zombie names_


so basically I am under the impression that for the bandits to not have zombie names, they need that line of code added to the mod/patch.

I have not made any patches in PZ before, for the record.

**Comment by theCrius** [comment]
Judging by the description there, it seems that first of all, the bandits mod need to be before Random Zombies in the mod load order.

That will allow the zombies to have the additional attribute `Bandit` that the line `getVariableBoolean("Bandit")` will retrieve.
What should happen next I guess is to skip the zombie entirely when it's marked as a bandit.

Now I see two main points of discussion here:
- This check will alter the balance of the distribution: as this is a flag inside specific zombies, the distribution (the percentages of zombies being a specific type) would not account for that when doing the initial calculation
- I am not sure how the logic of this mod might cause that issue described above (_"it will work but bandits will carry zombie names"_) as there is no line of code that interact in any way with the zombie name.

I suggest trying to initialize the Bandit mod before the Random Zombies and see if you encounter the issue with the names.
If you do, we can further work on it but, again, there is nothing touching the zombie names in this mod.

**Comment by malzuraga** [comment]
Hmm alright. I will try that out :) Thank you.

**Comment by malzuraga** [comment]
Ok so putting it above randomZ's breaks the bandit mod. Having Bandits below, and drastically reducing the zombie percentages in this mod, did not seem to affect anything. They might just be mostly incompatible in the end.

**Comment by theCrius** [comment]
I am not sure that I understand what you mean by "drastically reducing zombie percentage".

The zombie percentage will always add up to 100 (100%) so it cannot be drastically reduced.
I will have a look at how this Bandits mod's work over the next few days, just to see if there is some simple way to make them compatible. Can't promise anything though as I am quite busy at work at this time.

**Comment by malzuraga** [comment]
Cheers. Much appreciaed.

What I meant by reducing zombie percentages, was in sandbox settings, having totals end up at 50% for this mod.

**Comment by theCrius** [comment]
> Cheers. Much appreciaed.
> 
> What I meant by reducing zombie percentages, was in sandbox settings, having totals end up at 50% for this mod.

You should get an error if the sum of all percentages is less than 100%. If you didn't... well, that's a bug 😅
